aws/resource_aws_secretsmanager_secret_rotation.go

```diff
@@ -0,0 +1,208 @@
+		}
+
+		log.Printf("[DEBUG] Enabling Secrets Manager Secret rotation: %s", input)
+		err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+			output, err := conn.RotateSecret(input)
+			if err != nil {
+				// AccessDeniedException: Secrets Manager cannot invoke the specified Lambda function.
+				if isAWSErr(err, "AccessDeniedException", "") {
+					return resource.RetryableError(err)
+				}
+				return resource.NonRetryableError(err)
+			}
+
+			d.SetId(aws.StringValue(output.ARN))
+
+			return nil
+		})
+		if err != nil {
+			return fmt.Errorf("error enabling Secrets Manager Secret %q rotation: %s", d.Id(), err)
+		}
+	}
+
+	return resourceAwsSecretsManagerSecretRotationRead(d, meta)
+}
+
+func resourceAwsSecretsManagerSecretRotationRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).secretsmanagerconn
+
+	input := &secretsmanager.DescribeSecretInput{
+		SecretId: aws.String(d.Id()),
+	}
+
+	log.Printf("[DEBUG] Reading Secrets Manager Secret Rotation: %s", input)
+	output, err := conn.DescribeSecret(input)
+	if err != nil {
+		if isAWSErr(err, secretsmanager.ErrCodeResourceNotFoundException, "") {
+			log.Printf("[WARN] Secrets Manager Secret Rotation %q not found - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("error reading Secrets Manager Secret Rotation: %s", err)
+	}
+
+	d.Set("secret_id", d.Id())
+	d.Set("rotation_enabled", output.RotationEnabled)
+
+	if aws.BoolValue(output.RotationEnabled) {
+		d.Set("rotation_lambda_arn", output.RotationLambdaARN)
+		if err := d.Set("rotation_rules", flattenSecretsManagerRotationRules(output.RotationRules)); err != nil {
+			return fmt.Errorf("error setting rotation_rules: %s", err)
+		}
+	} else {
+		d.Set("rotation_lambda_arn", "")
+		d.Set("rotation_rules", []interface{}{})
+	}
+
+	return nil
+}
+
+func resourceAwsSecretsManagerSecretRotationUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).secretsmanagerconn
+	secretID := d.Get("secret_id").(string)
+
+	if d.HasChange("rotation_lambda_arn") || d.HasChange("rotation_rules") {
+		if v, ok := d.GetOk("rotation_lambda_arn"); ok && v.(string) != "" {
+			input := &secretsmanager.RotateSecretInput{
+				RotationLambdaARN: aws.String(v.(string)),
+				RotationRules:     expandSecretsManagerRotationRules(d.Get("rotation_rules").([]interface{})),
+				SecretId:          aws.String(secretID),
+			}
+
+			log.Printf("[DEBUG] Enabling Secrets Manager Secret Rotation: %s", input)
+			err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+				_, err := conn.RotateSecret(input)
+				if err != nil {
+					// AccessDeniedException: Secrets Manager cannot invoke the specified Lambda function.
+					if isAWSErr(err, "AccessDeniedException", "") {
+						return resource.RetryableError(err)
+					}
+					return resource.NonRetryableError(err)
+				}
+				return nil
+			})
+			if err != nil {
```

Comment:
To prevent issues where the API throttling requests during the entire timeout duration, this code should check against `isResourceTimeoutError()`, see also https://github.com/terraform-providers/terraform-provider-aws/issues/7873

```suggestion
			if isResourceTimeoutError(err) {
				_, err = conn.RotateSecret(input)
			}
			
			if err != nil {
```

Reply:
Awesome, I added this check in both the create and update functions.